core/engine/src/builtins/date/utils.rs

```diff
@@ -1,202 +1,730 @@
-/// The number of milliseconds in a second.
-pub(super) const MILLIS_PER_SECOND: i64 = 1000;
-/// The number of milliseconds in a minute.
-pub(super) const MILLIS_PER_MINUTE: i64 = MILLIS_PER_SECOND * 60;
-/// The number of milliseconds in an hour.
-pub(super) const MILLIS_PER_HOUR: i64 = MILLIS_PER_MINUTE * 60;
-/// The number of milliseconds in a day.
-pub(super) const MILLIS_PER_DAY: i64 = MILLIS_PER_HOUR * 24;
-
-// https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-time-values-and-time-range
+// Time-related Constants
 //
-// The smaller range supported by a time value as specified in this section is approximately -273,790 to 273,790
-// years relative to 1970.
-pub(super) const MIN_YEAR: i64 = -300_000;
-pub(super) const MAX_YEAR: i64 = -MIN_YEAR;
-pub(super) const MIN_MONTH: i64 = MIN_YEAR * 12;
-pub(super) const MAX_MONTH: i64 = MAX_YEAR * 12;
+// More info:
+// - [ECMAScript reference][spec]
+//
+// https://tc39.es/ecma262/#sec-time-related-constants
+
+// HoursPerDay = 24
+const HOURS_PER_DAY: f64 = 24.0;
+
+// MinutesPerHour = 60
+const MINUTES_PER_HOUR: f64 = 60.0;
+
+// SecondsPerMinute = 60
+const SECONDS_PER_MINUTE: f64 = 60.0;
+
+// msPerSecond = 1000𝔽
+const MS_PER_SECOND: f64 = 1000.0;
+
+// msPerMinute = 60000𝔽 = msPerSecond × 𝔽(SecondsPerMinute)
+pub(super) const MS_PER_MINUTE: f64 = MS_PER_SECOND * SECONDS_PER_MINUTE;
+
+// msPerHour = 3600000𝔽 = msPerMinute × 𝔽(MinutesPerHour)
+const MS_PER_HOUR: f64 = MS_PER_MINUTE * MINUTES_PER_HOUR;
+
+// msPerDay = 86400000𝔽 = msPerHour × 𝔽(HoursPerDay)
+const MS_PER_DAY: f64 = MS_PER_HOUR * HOURS_PER_DAY;
+
+/// Abstract operation `Day ( t )`
+///
+/// More info:
+/// - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-day
+pub(super) fn day(t: f64) -> f64 {
+    // 1. Return 𝔽(floor(ℝ(t / msPerDay))).
+    (t / MS_PER_DAY).floor()
+}
+
+/// Abstract operation `TimeWithinDay ( t )`
+///
+/// More info:
+/// - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-timewithinday
+pub(super) fn time_within_day(t: f64) -> f64 {
+    // 1. Return 𝔽(ℝ(t) modulo ℝ(msPerDay)).
+    modulo(t, MS_PER_DAY)
+}
+
+/// Abstract operation `DaysInYear ( y )`
+///
+/// More info:
+/// - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-daysinyear
+fn days_in_year(y: f64) -> u16 {
+    // 1. Let ry be ℝ(y).
+    let ry = y;
+
+    // 2. If (ry modulo 400) = 0, return 366𝔽.
+    if modulo(ry, 400.0) == 0.0 {
+        return 366;
+    }
+
+    // 3. If (ry modulo 100) = 0, return 365𝔽.
+    if modulo(ry, 100.0) == 0.0 {
+        return 365;
+    }
+
+    // 4. If (ry modulo 4) = 0, return 366𝔽.
+    if modulo(ry, 4.0) == 0.0 {
+        return 366;
+    }
+
+    // 5. Return 365𝔽.
+    365
+}
+
+/// Abstract operation `DayFromYear ( y )`
+///
+/// More info:
+/// - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-dayfromyear
+fn day_from_year(y: f64) -> f64 {
```

Comment:
I think this is implemented as a util in the `boa_temporal` crate (`epoch_day_number_for_year`). Would it be worth it to just reuse the code in this instance?